### PR TITLE
Added basic layout to build the Astropy documentation archive

### PR DESCRIPTION
**Pull request by astrofrog**
This is a companion PR to https://github.com/astropy/astropy/issues/7967 - see that issue for more details.

Note that if we merge this, it's not irreversible, and we still need to set up redirects and remove tagged versions from RTD. I'm just opening this in case anyone is interested, but I'll merge it to check if it works on RTD in a few days if no objections.

### CODE REVIEW

**Comment by pllim** [comment]
p.s. Not sure what's wrong with the bot, "skip-changelog-checks" should have instructed it to skip that check logic altogether. Does not appear to be working here.

**Comment by pllim** [comment]
p.p.s. Oh, you are not opening the PR against `master`. Maybe it went through a separate logic route that is not being handled properly.

File: docs/make.bat [hunk 1]
```diff
@@ -0,0 +1,35 @@
+@ECHO OFF
```
Comment (eteq) [inline docs/make.bat]: Minor, but maybe the `bat` file isn't necessary if we know we're building only on non-windows CI?

Comment (pllim) [reply docs/make.bat]: You mean, no one will *bat* an eye if it is gone? :wink:

**Comment by astrofrog** [comment]
As discussed with @eteq on slack, I've now removed the meta tags from the tgz archives (reverting back to the version as downloaded from RTD) and am now adding them inside the ``expand_archive`` sphinx extension here.

**Comment by astrofrog** [comment]
Since I've addressed @eteq's comment, I'm going to go ahead and merge this so I can start making sure it builds on RTD. I won't set up any redirects until we decide we want to do that for sure though, so if we change our minds at this point we can just remove the ``docs-archive`` branch.

**Pull request merged**